Add documentation for member ordering presets

grantwwu opened the pull request:
#### PR checklist

- [ ] Addresses an existing issue: #4051 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Overview of change:

Add a description to the member ordering rule presets.

#### Is there anything you'd like reviewers to focus on?

Why did the precommit hook add a ton of noise?

<!-- optional -->

#### CHANGELOG.md entry:
[no-log]
<!-- optional (example: "[new-rule] `arrow-return-shorthand`") -->
<!-- suggested tags: [new-rule], [new-rule-option], [new-fixer], [bugfix], [enhancement], [api], [rule-change], [no-log] -->

palantirtech commented:
Thanks for your interest in [palantir/tslint](https://github.com/palantir/tslint), @grantwwu! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

grantwwu commented:
I have rebased and fixed up my commits to align with the changes introduced in 85d87d1f12f9204dca957dfb8e99f74c99db8e3c.

I noticed that running `npm run prettier` still introduces some changes, incidentally.  @JoshuaKGoldberg is this intended?

JoshuaKGoldberg commented:
> running `npm run prettier` still introduces some changes

No ☹ not intended. What changes are you seeing?

grantwwu commented:
Just added some additional clarification regarding the presets.

@JoshuaKGoldberg this is what I'm seeing on my branch after `npm run prettier` (it seems 85aa8a70f6aeedb0bfb9f3d6c78443a332d69ba7 enabled fixing for orderedImportsRule.ts but didn't commit them?):

```
diff --git a/README.md b/README.md
index cdc4f034..fb51ef62 100644
--- a/README.md
+++ b/README.md
@@ -65,12 +65,12 @@ yarn test

 ## Creating a new release

-1. Bump the version number in `package.json` and `src/linter.ts`
-2. Add release notes in `CHANGELOG.md`
-    - Use `./scripts/generate-changelog.js` (after building it with `tsc -p scripts`) to generate the changelog diff. This script expects a [Github.com personal access token](https://github.com/settings/tokens) to exist at `~/github_token.txt` with "repo" permissions.
-3. Commit with message `Prepare release <version>`
-4. Push your branch to GitHub and make a PR
-5. Once your PR is merged, wait for the tests to pass on CircleCI for develop
-6. Create a "Release" on GitHub with the proper tag version and notes from the changelog.
-    - The tag should be identical to the version in `package.json`
-7. Run `yarn run publish:local`
+1.  Bump the version number in `package.json` and `src/linter.ts`
+2.  Add release notes in `CHANGELOG.md`
+    -   Use `./scripts/generate-changelog.js` (after building it with `tsc -p scripts`) to generate the changelog diff. This script expects a [Github.com personal access token](https://github.com/settings/tokens) to exist at `~/github_token.txt` with "repo" permissions.
+3.  Commit with message `Prepare release <version>`
+4.  Push your branch to GitHub and make a PR
+5.  Once your PR is merged, wait for the tests to pass on CircleCI for develop
+6.  Create a "Release" on GitHub with the proper tag version and notes from the changelog.
+    -   The tag should be identical to the version in `package.json`
+7.  Run `yarn run publish:local`
diff --git a/src/rules/orderedImportsRule.ts b/src/rules/orderedImportsRule.ts
index 9e2733c1..005283bd 100644
--- a/src/rules/orderedImportsRule.ts
+++ b/src/rules/orderedImportsRule.ts
@@ -182,7 +182,8 @@ function parseOptions(ruleArguments: any[]): Options {
         "import-sources-order": sources = "case-insensitive",
         "named-imports-order": named = "case-insensitive",
         "module-source-path": path = "full",
-    } = optionSet === undefined ? {} : optionSet;
+    } =
+        optionSet === undefined ? {} : optionSet;
     return {
         groupedImports: isGrouped,
         importSourcesOrderTransform: TRANSFORMS.get(sources)!,
```

JoshuaKGoldberg commented:
😕 I'm not seeing those changes locally after modifying `orderedImportsRule.ts`. Is your prettier version up-to-date as 1.14.3?

It probably makes sense to include them in this PR if they keep coming up.

grantwwu commented:
Changes no longer appearing after rerunning `yarn install`:thumbsup:

By the way, `npm i --package-lock-only && npm audit` states
```
found 17 vulnerabilities (15 low, 1 high, 1 critical)
  run `npm audit fix` to fix them, or `npm audit` for details
```
😱
